src/SDL.c

```diff
@@ -107,6 +107,52 @@ SDL_NORETURN void SDL_ExitProcess(int exitcode)
 #endif
 }
 
+/* App metadata */
+static SDL_PropertiesID SDL_app_metadata_properties = 0;
+
+int SDL_SetAppMetadata(const char *appname, const char *appversion, const char *appidentifier)
+{
+    const SDL_PropertiesID props = SDL_CreateProperties();
+
+    if (appname && *appname) {
+        SDL_SetStringProperty(props, SDL_PROP_APP_METADATA_NAME_STRING, appname);
+    }
+    if (appversion && *appversion) {
+        SDL_SetStringProperty(props, SDL_PROP_APP_METADATA_VERSION_STRING, appversion);
+    }
+    if (appidentifier && *appidentifier) {
+        SDL_SetStringProperty(props, SDL_PROP_APP_METADATA_IDENTIFIER_STRING, appidentifier);
+    }
+
+    const int retval = SDL_SetAppMetadataWithProperties(props);
```

Comment:
I don't think we have had any formal discussion about this, but can we just use const for pointers in the SDL codebase?

Reply:
Now that we can declare variables in the middle of a block, it's super useful to be able to articulate "this value isn't meant to change, and the compiler should stop me if I accidentally assign to it later." Do we really want to have a policy against that?

This particular variable isn't a big deal, though.

Reply:
Sure, that's fair. Maybe we should do a pass to make everything const where possible?

Reply:
Never mind, we still need to maintain mergability with SDL2.

Reply:
Yeah, I think it's something that happens as code changes, even if SDL2 wasn't an issue.